ecs-cli/modules/cli/compose/entity/service/service_test.go

```diff
@@ -596,6 +596,57 @@ func TestUpdateExistingServiceWithDesiredCountOverOne(t *testing.T) {
 	upServiceWithNewTaskDefTest(t, cliContext, &config.CommandConfig{}, &utils.ECSParams{}, expectedInput, existingService)
 }
 
+func TestCreateWithServiceDiscovery(t *testing.T) {
+	sdsARN := "arn:aws:servicediscovery:eu-west-1:11111111111:service/srv-clydelovespudding"
+
+	flagSet := flag.NewFlagSet("ecs-cli-up", 0)
+	flagSet.Bool(flags.EnableServiceDiscoveryFlag, true, "")
+
+	servicediscoveryCreate = func(c *cli.Context, networkMode, serviceName, clusterName string) (*string, error) {
```

Comment:
Don't we need to reset this mock to be what it was? (Goes back to discussion on setting global state from #565)

Reply:
Yes! Good catch, I'll fix this.